pkg/apis/operator/v1/amazoncloudintegration_types.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	AmazonCloudIntegrationStatusReady = "Ready"
+)
+
+// MetadataAccessAllowedType
+type MetadataAccessAllowedType string
+
+const (
+	MetadataAccessAllowed MetadataAccessAllowedType = "Allowed"
+	MetadataAccessDenied  MetadataAccessAllowedType = "Denied"
+)
+
+// AmazonCloudIntegrationSpec defines the desired state of AmazonCloudIntegration
+// +k8s:openapi-gen=true
+type AmazonCloudIntegrationSpec struct {
+	// DefaultPodMetadataAccess defines what the default behavior will be for accessing
+	// metadata from a pod.
```

Comment:
Might be useful to be a bit more explicit here since we don't want to conflate the aws metadata with k8s metadata. 
```suggestion
	// the AWS metadata service from a pod.
```